src/App.tsx

```diff
@@ -8,7 +8,64 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { getTodos } from './api';
+
+import { Todo } from './types/Todo';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [isLoad, setIsLoad] = useState(false);
+  const [query, setQuery] = useState('');
+  const [filterBy, setFilterBy] = useState('all');
+
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+
+  const loadTodos = async () => {
+    const todos = await getTodos();
+
+    setIsLoad(true);
+
+    setTodos(todos);
+  };
+
+  const filteredTodos = todos.filter((todo: Todo) => todo.title.toLocaleLowerCase().includes(query.toLocaleLowerCase()));
+  let visibleTodos = filteredTodos;
+
+  const setVisibleTodos = () => {
+    switch (filterBy) {
+      case 'all':
+        return filteredTodos;
+      case 'completed':
+        return visibleTodos = filteredTodos.filter((todo) => todo.completed);
+
+      case 'active':
+        return visibleTodos = filteredTodos.filter((todo) => !todo.completed);
+
+      default:
+        return;
+    }
+  };
+
+  const onHideModal = () => {
+    setSelectedTodo(null);
+  };
+
+  const handleTodoSelect = (todo: Todo) => {
+    setSelectedTodo(todo);
+  };
+
+  const handleQuatyChange = (value: string) => {
+    setQuery(value);
+  };
+
+  const handleFilteredBy = (value: string) => {
+    setFilterBy(value);
+  };
+
+  setVisibleTodos();
+
+  loadTodos();
```

Comment:
Dont call `functions` each `render`, move it to `handler function `or `useEffect`